### DISCUSSION TITLE
Add rich text to a cell

### DISCUSSION THREAD

**Discussion by Yosi096** [PHPOffice/PhpSpreadsheet#2916] [Q&A] [answered]
how to use richText in looping (for) . use in documentation not working. thanks.

for ($i = 0; $i < count(array_keys($Data)); $i++) {	
     $richText = new RichText();
     $payable = $richText->createTextRun($Data[$i]->Name);
     $payable->getFont()->setBold(true);
     $payable->getFont()->setColor(new Color(Color::COLOR_DARKGREEN));

     $sheet->setCellValue('A' . $x, "\n" . $num);
     $sheet->setCellValue('B' . $x, "\n" . $richText . "\n" . $Data[$i]->Age);
     $sheet->setCellValue('C' . $x, "\n" . $Address);
}

**Comment by MarkBaker** [comment]
I'm not sure exactly what you're trying to do here, because you could just as easily set the cell styles rather than using Rich Text; but what exactly is the problem that you're having? What do you expect? And what do you get? And what is `$x`?

**Reply by Yosi096** [reply]
I want in cell B only the name is bold and the age is not. $x is the number of the column.

**Comment by MarkBaker** [comment, marked as answer]
Then you need one rich text object that has two runs, one with the name and the styling, the second run with the age and no styling; and you write that rich text object to the cell... you're trying to concatenate a rich text object with plain text (which will result in a standard PHP string of plain text) and then write that to the cell
A Excel cell can't contain both a plain text string **_and_** a rich text object, it can contain either plain text **_or_** rich text

But, a rich text object can contain a mixture of styled text and unstyled text
```php
$richText = new RichText();
$payable = $richText->createTextRun($Data[$i]->Name);
$payable->getFont()->setBold(true);
$payable->getFont()->setColor(new Color(Color::COLOR_DARKGREEN));
$richText->createText("\n" . $Data[$i]->Age);
```
